### ISSUE TITLE
Wrong mRest positioning

### ISSUE THREAD

**Issue by rettinghaus** [rism-ch/verovio#1728]
There seems to be a weird rounding issue with positioning mRests with some scaling factors. See measure rest positions in staves 1, 2 and 4.
![barline](https://user-images.githubusercontent.com/7693447/95188913-6fc7fe80-07cd-11eb-9d03-fc91f6e2b6d7.png)

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title></title>
            </titleStmt>
            <pubStmt>
                <date></date>
            </pubStmt>
        </fileDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef label="feature-example">
                      <meterSig unit="4" count="4"/>
                        <staffGrp>
                            <staffGrp symbol="bracketsq" bar.thru="true" xml:id="System">
                                <staffDef n="1" clef.line="2" clef.shape="G" lines="5" scale="75%"/>
                                <staffGrp symbol="brace" xml:id="klammern">
                                    <staffDef n="2" clef.line="2" clef.shape="G" lines="5" scale="63%"/>
                                    <staffDef n="3" clef.line="2" clef.shape="G" lines="5"/>
                                </staffGrp>
                            </staffGrp>
                            <staffDef n="4" clef.line="2" clef.shape="G" lines="5" scale="51%"/>
                        </staffGrp>
                    </scoreDef>
                    <section>
                        <measure n="4" right="rptstart">
                            <staff n="1">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="2">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="3">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="4">
                                <layer n="1"><mRest/></layer>
                            </staff>
                        </measure>
                        <measure n="5" right="rptboth">
                            <staff n="1">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="2">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="3">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="4">
                                <layer n="1"><mRest/></layer>
                            </staff>
                        </measure>
                        <measure n="6" right="rptend">
                            <staff n="1">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="2">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="3">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="4">
                                <layer n="1"><mRest/></layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
Maybe a rounding issue, but probably more likely to be the staff size being always 100% when calculating the position of scaled rests - or something like that.

**Comment by rettinghaus** [comment]
I noticed that this doesn't seem to work properly: 
`Staff *staff = dynamic_cast<Staff *>(barLine->GetFirstAncestor(STAFF));`
Maybe then connected in some way …

**Comment by lpugin** [comment]
Where?

**Comment by rettinghaus** [comment]
In `View::DrawBarLine`.

**Comment by lpugin** [comment]
OK, I will look for it. Next time I should remind myself not to forget to ask for the line number... ;-)

**Comment by lpugin** [comment]
Fixed with 25eefb9b4258debcdca54fc82c5d62b939ef2808

![image](https://user-images.githubusercontent.com/689412/95207519-3f418e00-07e8-11eb-99bd-4c573bc1bbc2.png)

**Comment by rettinghaus** [comment]
Thanks!
